### PR TITLE
Validate file names

### PR DESCRIPTION
**Pull request by DanielElisenberg**
Validate filenames more explicitly and return a validation error like the rest of the validation results.
Fixed some minor bugs where variables could end up not being declare before the finally clause

### CODE REVIEW

**Review by haraldsvik — DISMISSED** [review]
Looks good!

File: microdata_tools/validation/adapter/local_storage.py [hunk 1]
```diff
@@ -41,10 +67,13 @@ def resolve_working_directory(
         return generated_working_directory, True
 
 
+logger.warning("This function will be deprecated in a future release")
```
Comment (pawbu) [inline microdata_tools/validation/adapter/local_storage.py:70]: No indentation? Module-level statement?

**Review by pawbu — APPROVED** [review]
Looks good. Version 0.11.0 was just released, so this PR needs a new one 👍

**Review by pawbu — APPROVED** [review]
💯

**Pull request merged**